src/jit/rangecheck.cpp

```diff
@@ -439,6 +445,40 @@ bool RangeCheck::IsMonotonicallyIncreasing(GenTreePtr expr)
     return false;
 }
 
+// Given a lclvar use, try to find the lclvar's defining assignment and its containing block.
+GenTreeOp* RangeCheck::GetSsaDefAsg(GenTreeLclVarCommon* lclUse, BasicBlock** asgBlock)
+{
+    unsigned ssaNum = lclUse->GetSsaNum();
+
+    if (ssaNum == SsaConfig::RESERVED_SSA_NUM)
+    {
+        return nullptr;
+    }
+
+    LclSsaVarDsc* ssaData = m_pCompiler->lvaTable[lclUse->GetLclNum()].GetPerSsaData(ssaNum);
+    GenTree*      lclDef  = ssaData->m_defLoc.m_tree;
+
+    if (lclDef == nullptr)
+    {
+        return nullptr;
+    }
+
+    // We have the def node but we also need the assignment node to get its source.
+    // gtGetParent can be used to get the assignment node but it's rather expensive
+    // and not strictly necessary here, there shouldn't be any other node between
+    // the assignment node and its destination node.
+    GenTree* asg = lclDef->gtNext;
+
+    if (!asg->OperIsAssignment())
+    {
+        return nullptr;
+    }
+
```

Comment:
Check/assert here that `lclDef` is in fact the LHS of `asg`?

Reply:
Hrm, it seems that I keep forgetting this, I'll add a check to be sure. Though it seems impossible for `lclDef` not to be the LHS of an ASG that follows it you can never be sure about certain things in the JIT world...